Cache/Config.lean

```diff
@@ -0,0 +1,84 @@
+import Lean.Data.RBMap
+import Lean.Data.Json
+
+open System (FilePath mkFilePath)
+open Lean (Json RBMap)
+
+namespace Cache
+
+abbrev PkgDirs := Lean.RBMap String FilePath compare
+
+structure Config where
+  url : String
+  head : FilePath
+  rootDir : FilePath
+  tokenEnvVar : String
+  pkgDirs: PkgDirs
+
+namespace Config
+
+@[inline] def getBuildDir (cfg : Config) (k : String) : Option FilePath :=
+  cfg.pkgDirs.find? k
+
+def FILE : FilePath :=
+  ⟨"cache-config.json"⟩
+
+def stdTemplate : String :=
+"{
+  \"url\":\"https://lakecache.blob.core.windows.net/std4\",
+  \"head\":\"Std.lean\",
+  \"rootDir\":\"lake-packages/std\",
+  \"tokenEnvVar\":\"STD_CACHE_SAS\",
+  \"pkgDirs\":
+  {
+    \"Std\":\"lake-packages/std\"
+  }
+}\n"
+
+def mathlibTemplate : String :=
+"{
+  \"url\":\"https://lakecache.blob.core.windows.net/mathlib4\",
+  \"head\":\"Mathlib.lean\",
+  \"rootDir\":\"lake-packages/mathlib\",
+  \"tokenEnvVar\":\"MATHLIB_CACHE_SAS\",
+  \"pkgDirs\":
+  {
+    \"Mathlib\":\"lake-packages/mathlib\",
+    \"Aesop\":\"lake-packages/aesop\",
+    \"Qq\":\"lake-packages/Qq\",
+    \"Std\":\"lake-packages/std\"
+  }
+}\n"
+
+def check : IO Bool := do
+  if ← FILE.pathExists then return true else
+  IO.println s!"Which cache config template to use?\n(0) Std\n(1) Mathlib"
```

Comment:
Ideally, it would pick automatically pick 1 if mathlib is a dependency, and 0 otherwise.

Reply:
This should be easy if we can figure out the dependent packages.  See above.